### PR TITLE
feat: abort patching process at any time

### PR DESCRIPTION
**Pull request by TheAabedKhan**
This PR aims to allow users to abort the patching process whenever they want so that they can include/exclude patches or change the app without having to restart the manager.

Make sure the following tasks are completed. Without this, the patching process can be aborted but cannot be started again without restarting the manager.

- [x]  Merged https://github.com/ReVanced/revanced-patcher/pull/211
- [x]  Bumped ReVanced Patcher version in the manager to the version with that fix.

### CODE REVIEW

**Comment by oSumAtrIX** [comment]
Does this kill the patcher thread?

**Comment by TheAabedKhan** [comment]
> Does this kill the patcher thread?

Yes. The user can only exit the page after the thread is stopped safely.

File: android/app/src/main/kotlin/app/revanced/manager/flutter/MainActivity.kt [hunk 1]
```diff
@@ -101,170 +108,238 @@ class MainActivity : FlutterActivity() {
...
+                                "header" to "Signing apk...",
+                                "log" to ""
+                            )
                         )
-                    )
-                }
+                    }
 
-                // Signer("ReVanced", "s3cur3p@ssw0rd").signApk(patchedFile, outFile, keyStoreFile)
+                    // Signer("ReVanced", "s3cur3p@ssw0rd").signApk(patchedFile, outFile, keyStoreFile)
```
Comment (Ushie) [inline android/app/src/main/kotlin/app/revanced/manager/flutter/MainActivity.kt]: you could delete this comment

File: android/app/src/main/kotlin/app/revanced/manager/flutter/MainActivity.kt [hunk 2]
```diff
@@ -247,7 +314,11 @@ class MainActivity : FlutterActivity() {
                 // Signer("ReVanced", "s3cur3p@ssw0rd").signApk(patchedFile, outFile, keyStoreFile)
```
Comment (Ushie) [inline android/app/src/main/kotlin/app/revanced/manager/flutter/MainActivity.kt]: You can delete this comment

File: android/app/src/main/kotlin/app/revanced/manager/flutter/MainActivity.kt [hunk 3]
```diff
@@ -27,6 +27,8 @@ private const val INSTALLER_CHANNEL = "app.revanced.manager.flutter/installer"
 class MainActivity : FlutterActivity() {
     private val handler = Handler(Looper.getMainLooper())
     private lateinit var installerChannel: MethodChannel
+    private var run: Boolean = true
```
Comment (Ushie) [inline android/app/src/main/kotlin/app/revanced/manager/flutter/MainActivity.kt:30]:
```suggestion
    private var cancel: Boolean = true
```

Maybe rename this to cancel so the if statements are more straightforward with their purpose?

**Review by Ushie — CHANGES_REQUESTED** [review]

**Comment by ILoveOpenSourceApplications** [comment]
Has this been reverted or something as I cannot see the option to abort/cancel after clicking the patch button.

**Comment by oSumAtrIX** [comment]
You can cancel by pressing back.

**Pull request merged**